www-apps/redmine/redmine-3.3.3.ebuild

```diff
@@ -0,0 +1,218 @@
+#	dev-ruby/yard
+#	test? (
+#		>=dev-ruby/shoulda-3.3.2
+#		>=dev-ruby/mocha-0.13.3
+#		>=dev-ruby/capybara-2.0.0
+#		<dev-ruby/nokogiri-1.6.0
+#	)"
+
+REDMINE_DIR="/var/lib/${PN}"
+
+pkg_setup() {
+	enewgroup redmine
+	enewuser redmine -1 -1 "${REDMINE_DIR}" redmine
+}
+
+all_ruby_prepare() {
+	rm -r log files/delete.me Gemfile || die
+
+	# bug #406605
+	rm .{git,hg}ignore || die
+
+	cat > ${T}/50${PN} <<-EOF
+		CONFIG_PROTECT="${EROOT%/}${REDMINE_DIR}/config"
+		CONFIG_PROTECT_MASK="${EROOT%/}${REDMINE_DIR}/config/locales ${EROOT%/}${REDMINE_DIR}/config/settings.yml"
+	EOF
+
+	# remove ldap staff module if disabled to avoid #413779
+	use ldap || rm app/models/auth_source_ldap.rb || die
+
+	epatch "${FILESDIR}/${P}_requires.patch"
+}
+
+all_ruby_install() {
+	dodoc doc/{CHANGELOG,INSTALL,README_FOR_APP,RUNNING_TESTS,UPGRADING}
+	dodoc README.rdoc
+	rm -r doc appveyor.yml CONTRIBUTING.md README.rdoc || die
+
+	keepdir /var/log/${PN}
+
+	insinto "${REDMINE_DIR}"
+	doins -r .
+	insinto "${REDMINE_DIR}/config"
+	doins "${FILESDIR}/additional_environment.rb"
+	keepdir "${REDMINE_DIR}/files"
+	keepdir "${REDMINE_DIR}/public/plugin_assets"
+
+	fowners -R redmine:redmine \
+		"${REDMINE_DIR}/config" \
+		"${REDMINE_DIR}/files" \
+		"${REDMINE_DIR}/public/plugin_assets" \
+		"${REDMINE_DIR}/tmp" \
+		/var/log/${PN}
+
+	fowners redmine:redmine "${REDMINE_DIR}"
+
+	# protect sensitive data, see bug #406605
+	fperms -R go-rwx \
+		"${REDMINE_DIR}/config" \
+		"${REDMINE_DIR}/files" \
+		"${REDMINE_DIR}/tmp" \
+		/var/log/${PN}
+
+	if use passenger; then
+		has_apache
+		insinto "${APACHE_VHOSTS_CONFDIR}"
+		doins "${FILESDIR}/10_redmine_vhost.conf"
+	fi
+
+	# logrotate
+	insinto /etc/logrotate.d
+	newins "${FILESDIR}"/redmine.logrotate redmine
+
+	newconfd "${FILESDIR}/${PN}.confd" ${PN}
+	newinitd "${FILESDIR}/${PN}-3.initd" ${PN}
+	doenvd "${T}/50${PN}"
+}
+
+pkg_postinst() {
+	einfo
+	if [ -e "${EROOT%/}${REDMINE_DIR}/config/initializers/session_store.rb" ] || [ -e "${EROOT%/}${REDMINE_DIR}/config/initializers/secret_token.rb" ]; then
+		elog "Execute the following command to upgrade environment:"
+		elog
+		elog "# emerge --config \"=${CATEGORY}/${PF}\""
+		elog
+		elog "For upgrade instructions take a look at:"
+		elog "http://www.redmine.org/wiki/redmine/RedmineUpgrade"
+	else
+		elog "Execute the following command to initialize environment:"
+		elog
+		elog "# cd ${EROOT%/}${REDMINE_DIR}"
+		elog "# cp config/database.yml.example config/database.yml"
+		elog "# \${EDITOR} config/database.yml"
+		elog "# chown redmine:redmine config/database.yml"
+		elog "# emerge --config \"=${CATEGORY}/${PF}\""
+		elog
+		elog "Installation notes are at official site"
+		elog "http://www.redmine.org/wiki/redmine/RedmineInstall"
+	fi
+	einfo
+}
+
+pkg_config() {
+	if [ ! -e "${EROOT%/}${REDMINE_DIR}/config/database.yml" ]; then
+		eerror "Copy ${EROOT%/}${REDMINE_DIR}/config/database.yml.example to ${EROOT%/}${REDMINE_DIR}/config/database.yml"
```

Comment:
I think this line can end up veery long. I'd suggest wrapping at 80, or approximate 80 ;-).